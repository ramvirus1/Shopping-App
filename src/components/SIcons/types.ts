export interface ISIcons {
    name: any;
    size: number;
    color?: string;
}