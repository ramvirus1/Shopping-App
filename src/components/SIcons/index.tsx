import Ionicons from '@expo/vector-icons/Ionicons';
import { scale, verticalScale, moderateScale } from 'react-native-size-matters';

const SIcons = ({ 
    name, 
    size, 
    color = "black"
}): JSX.Element => {
    return <Ionicons name={name} size={scale(size)} color={color} />
};

export default SIcons;