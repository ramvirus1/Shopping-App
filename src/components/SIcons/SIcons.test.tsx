import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import SIcons from "./index";

const props = {
    name: "remove",
    size: 32,
};

describe('SIcons', () => {
  it('should render', () => {
    const snapshot = renderer.create(<SIcons {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});