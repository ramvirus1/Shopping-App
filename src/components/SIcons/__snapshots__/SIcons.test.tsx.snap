// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SIcons should render 1`] = `<Text />`;
